src/utilities/logUserPermissionChangeByAccount.js

```diff
@@ -19,13 +22,26 @@ const logUserPermissionChangeByAccount = async (req) => {
     if (document) {
       const docPermissions = Array.isArray(document.permissions) ? document.permissions : [];
       // no new changes in permissions list from last update
-      if (JSON.stringify(docPermissions.sort()) === JSON.stringify(Permissions.sort())) {
+      if (JSON.stringify(docPermissions.sort()) === JSON.stringify(changedPermissions.sort())) {
         return;
       }
-      permissionsRemoved = docPermissions.filter((item) => !Permissions.includes(item));
-      permissionsAdded = Permissions.filter((item) => !docPermissions.includes(item));
+      permissionsRemoved = [
+        ...removedPermissions.filter((item) => !docPermissions.includes(item)), //saves new removed defaults
+        ...docPermissions.filter(
+          // saves changes of only removed non-default role permissions for user
+          (item) => !defaultPermissions.includes(item) && !Permissions.includes(item),
+        ),
+      ];
+      permissionsAdded = [
+        ...Permissions.filter((item) => !docPermissions.includes(item)), // saves new added permissions
+        ...docPermissions.filter(
+          // saves changes of only removed default permissions being added back
+          (item) => defaultPermissions.includes(item) && !removedPermissions.includes(item),
+        ),
+      ];
```

Comment:
Adding/removing a permission to a user, then mirroring that change to the role will keep showing that change as new with every update.

```suggestion
      permissionsRemoved = [
        ...removedPermissions.filter((item) => !prevRemovedPermissions.includes(item)), //saves new removed defaults
        ...prevAddedPermissions.filter((item) =>!Permissions.includes(item)), //removed user permissions
      ];
      permissionsAdded = [
        ...Permissions.filter((item) => !prevAddedPermissions.includes(item)), // saves new added permissions
        ...prevRemovedPermissions.filter((item) => !removedPermissions.includes(item)), //removed permissions added back
      ];
```

Reply:
I understand that concern, I'll have to take a look into how changing the role of a user with added permissions or removed role permissions affects things with the logs before I can see if it causes issues, and then try implementing the changes you suggested to see if that helps resolve it. 

Currently though, when I try modifying the role of my alt manager account on the development live app or locally, I receive an error. This is when I open the profile of my alt, while on my owner account, scroll to their Role, open the role menu and select administrator or volunteer, just to try changing for the changes in permissions, and receive the error I shared below. Does this occur for you? Otherwise if its just my end, I am unsure how to resolve it so I can test the potential issue with role changes with my code. Thank you for bringing up the potential issues though!

<img width="501" height="217" alt="image_2025-08-31_211439223" src="https://github.com/user-attachments/assets/2b1efbb4-bea6-41e7-af91-2ad02d00e5c6" />

Reply:
Nevermind, it was an error that occurred during work on another PR, after deleting the incorrect data, I was able to change the role of my alt to test your concerns. You are right that role changes does impact the change log, and your suggested code to handle permissions added and remove does seem better, but in the case of prevAddedPermissions, if we add a permission, say 'Edit Task' to a user, then updated their role to be higher, which includes 'Edit Task' as a default, if a new change log is made for the user, then ...prevAddedPermissions.filter((item) =>!Permissions.includes(item)) in setting permissionsRemoved would be triggered, as I checked, and on the frontend ui for the permission list, through console logs, when I checked the permissions of my alt, its permissions were set to the role's default, so the respective permissions and removedPermissions become blank at first. So with 'Edit Task', it would no be in Permissions if it becomes a default permission due to role, thus being counted as one to be listed under permissionsRemoved. So I should also include a check to see if the item in question, is listed in the role's default list?